traffic_router/core/src/test/java/com/comcast/cdn/traffic_control/traffic_router/core/external/ConsistentHashTest.java

```diff
@@ -258,4 +263,172 @@ public void itUsesBypassFiltersWithDeliveryServiceSteering() throws Exception {
 			if (response != null) response.close();
 		}
 	}
+
+	@Test
+	public void itUsesRegexToStandardizeRequestPath() throws Exception {
+		CloseableHttpResponse response = null;
+
+		try {
+			String requestPath = URLEncoder.encode("/some/path/thing.m3u8", "UTF-8");
+			String encodedConsistentHashRegex = URLEncoder.encode(consistentHashRegex, "UTF-8");
+			HttpGet httpGet = new HttpGet("http://localhost:3333/crs/consistenthash/patternbased/regex?regex=" + encodedConsistentHashRegex + "&requestPath=" + requestPath);
+
+			response = closeableHttpClient.execute(httpGet);
+
+			assertThat("Expected to get 200 response from /consistenthash/patternbased/regex endpoint", response.getStatusLine().getStatusCode(), equalTo(200));
+
+			ObjectMapper objectMapper = new ObjectMapper(new JsonFactory());
+			JsonNode resp = objectMapper.readTree(EntityUtils.toString(response.getEntity()));
+			String resultingPathToConsistentHash = resp.get("resultingPathToConsistentHash").asText();
+
+			requestPath = URLEncoder.encode("/other/path/other_thing.m3u8", "UTF-8");
+			httpGet = new HttpGet("http://localhost:3333/crs/consistenthash/patternbased/regex?regex=" + encodedConsistentHashRegex + "&requestPath=" + requestPath);
+
+			response = closeableHttpClient.execute(httpGet);
+
+			resp = objectMapper.readTree(EntityUtils.toString(response.getEntity()));
+
+			assertThat(JsonUtils.optString(resp, "resultingPathToConsistentHash"),equalTo(resultingPathToConsistentHash));
+		} finally {
+			if (response != null) response.close();
```

Comment:
nit/question: I _think_ we can use https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html here since `CloseableHttpResponse` implements `java.io.Closeable`. That might clean these test cases up slightly.

Reply:
so I tried adjusting a test case to use:
`try (CloseableHttpResponse response = closeableHttpClient.execute(httpGet)){ 
...
}`
but it looks like a resource declared with a try statement is final, and that's not ideal here since I'm reusing the `CloseableHttpResponse` object for multiple requests in each test case

Reply:
I could add a try-with-resource statement for each request, but I think that's going the opposite way in terms of cleanliness

Reply:
yeah, thanks for trying it out. If it makes it messier we shouldn't do it.